### PR TITLE
v1.29 Support

### PR DESCRIPTION
**Pull request by ebauman**
Initially this PR began as support for v1.28. There were changes in k8s.io/apiserver that required mink to pass a kubeconfig to some setup calls:

https://github.com/kubernetes/apiserver/blob/8f40ac3bd1be6778545893b56fad7863ce196c5c/pkg/server/options/recommended.go#L135-L141

I soon discovered that these changes however were incompatible with just a dummy kubeconfig file. Specifically even if one did pass a dummy kubeconfig, there was still remaining code related to api fairness that wasn't easily worked around:

https://github.com/kubernetes/apiserver/blob/8f40ac3bd1be6778545893b56fad7863ce196c5c/pkg/server/options/recommended.go#L147-L158

However if we hold off getting 1.28 support until, ironically, 1.29, we can sidestep this issue entirely. 1.29 appears to have removed the APF stuff that was problematic. Code as of v0.29.0-alpha.3: 

https://github.com/kubernetes/apiserver/blob/10c70bebf96a41cfc1fb721c80a668d648cfeb0c/pkg/server/options/recommended.go#L98-L143

Verified that passing a dummy kubeconfig (implemented in `pkg/server/server.go`) satisfies at least topically the `ApplyTo` call. 

v0.29.0 introduced an "APIServerID" that needs to be passed to an encryption setup call. I have modified the one method in mink that uses this - it doesn't appear this is used in runtime currently but I cannot see anything beyond public repos so YMMV. 

Leaving this in draft state until v1.29 is released and we can confirm final changes to upstream.

### CODE REVIEW

**Comment by ebauman** [comment]
Worth noting here that v1.29 is scheduled for early December so waiting a few weeks isn't too rough.

File: pkg/server/server.go [hunk 1]
```diff
@@ -218,6 +220,34 @@ func (s *Server) Run(ctx context.Context) error {
 	return nil
 }
 
+func generateDummyKubeconfig() *rest.Config {
+	return &rest.Config{
```
Comment (ibuildthecloud) [inline pkg/server/server.go:224]: Can this just be `&rest.Config{}` is there any benefit to all the fields? Those seem like all empty values.

Comment (ibuildthecloud) [reply pkg/server/server.go:224]: I'm fine with the method being there it indicates the intent, just seems like it can be a single line

Comment (ebauman) [reply pkg/server/server.go]: Fixed.

**Pull request merged**